### PR TITLE
feat(sdk): allow to unset `content-type` by passing `null` as value

### PR DESCRIPTION
**Pull request by emmenko**
Ref #2469

### CODE REVIEW

File: .changeset/clean-chairs-unite.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+'@commercetools-frontend/sdk': minor
+---
+
+The SDK request payload (body) now allows to pass something other than a `string` or `Json` objects.
```
Comment (tdeekens) [inline .changeset/clean-chairs-unite.md:5]: Nit: to me it's JSON. Also if the type name is different. I think in our docs it's always all capital letters.

File: website/src/content/development/data-fetching.mdx [hunk 2]
```diff
@@ -279,3 +279,27 @@ The available proxy targets are exposed as a `MC_API_PROXY_TARGETS` constant in
...
 
 </Warning>
+
+<!-- TODO: ### Using your own HTTP client -->
+
+### Using file uploads
+
+In case you need to deal with file uploads, you will probably use [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData).
+
+The `FormData` object can then be sent as the request HTTP body.
```
Comment (tdeekens) [inline website/src/content/development/data-fetching.mdx:289]: I think it should be send not sent here.

File: website/src/content/development/data-fetching.mdx [hunk 3]
```diff
@@ -279,3 +279,27 @@ The available proxy targets are exposed as a `MC_API_PROXY_TARGETS` constant in
...
+
+One important thing is to properly configure the `Content-Type` HTTP header, which usually must be set to `multipart/form-data`. However, you also need to provide a `boundary` [directive](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type#directives).
```
Comment (tdeekens) [inline website/src/content/development/data-fetching.mdx:291]: Additionally it's important to...

One important thing is odd wording to me.

File: website/src/content/development/data-fetching.mdx [hunk 4]
```diff
@@ -279,3 +279,27 @@ The available proxy targets are exposed as a `MC_API_PROXY_TARGETS` constant in
...
+
+The best way to properly configure it, is to let the browser do it. You do so by not providing the `Content-Type` HTTP header in the request. The browser will automatically set one.
```
Comment (tdeekens) [inline website/src/content/development/data-fetching.mdx:293]: In this case the browser can infer the Content-Type correctly. We allow the browser to do so by not...

Then omit last sentence. 

Sorry am on my phone.

File: website/src/content/development/data-fetching.mdx (hunk 2 above)
Comment (ddouglasz) [reply website/src/content/development/data-fetching.mdx:289]: I know this sounds weird @tdeekens, but I think "sent" is actually the correct tense in this case.

Comment (tdeekens) [reply website/src/content/development/data-fetching.mdx:289]: Oh ok. It really felt against my intuition. Thanks for clarifying.

**Review by tdeekens — APPROVED** [review]

**Review by CarlosCortizasCT — APPROVED** [review]
LGTM

**Pull request merged**